### PR TITLE
Add update activities endpoint management

### PR DESCRIPTION
**Pull request by reifcode**
Adds the `UpdateActivities(...*Activity)` method to feeds, so they can be properly updated (https://getstream.io/docs_rest/#activities).

Also cleans up the code a little bit and fixes a potential nil pointer exception when signing JWT requests.

### CODE REVIEW

File: feed.go [hunk 1]
```diff
@@ -184,6 +184,25 @@ func (f *baseFeed) RemoveActivityByForeignID(foreignId string) error {
 	})
 }
 
+func (f *baseFeed) UpdateActivities(activities ...*Activity) error {
+	for _, a := range activities {
```
Comment (tbarbugli) [inline feed.go]: we already do this check server-side

Comment (reifcode) [reply feed.go]: Yup, realised it while working on the last PR, moving this away

File: feed_test.go [hunk 2]
```diff
@@ -84,3 +89,84 @@ func TestNotificationFeedBasic(t *testing.T) {
 		t.Fatal()
 	}
 }
+
+func TestUpdateActivities(t *testing.T) {
```
Comment (JelteF) [inline feed_test.go]: Test should go int `feed_flat_test.go` with all the other integration tests and it should use helpers such as `PreTestSetup` and `getFlatFeed`.

File: feed_test.go [hunk 3]
```diff
@@ -84,3 +89,84 @@ func TestNotificationFeedBasic(t *testing.T) {
...
+					Actor:     "bob",
+					Verb:      "verb",
+					Object:    "object",
+				},
+			},
+			shouldError: true,
+		},
+		{
+			activities:  []*Activity{},
+			shouldError: true,
```
Comment (JelteF) [inline feed_test.go]: The error tests should be part of QA, not part of the client tests. Those should just do happy path.

File: feed_test.go [hunk 4]
```diff
@@ -84,3 +89,84 @@ func TestNotificationFeedBasic(t *testing.T) {
...
+		{
+			activities: []*Activity{
+				&Activity{
+					ForeignID: "alice:123",
+					TimeStamp: &tt,
+					Actor:     "alice",
+					Verb:      "verb",
+					Object:    "object",
+				},
+			},
```
Comment (JelteF) [inline feed_test.go]: I might be missing something, but this test seems to be a duplicate of the first one.

File: feed_test.go (hunk 2 above)
Comment (reifcode) [reply feed_test.go]: this is not a flat feed-only test. it's testing a method of `baseFeed`, therefore should be tested where the other `baseFeed` methods are tested. I agree on the `PreTestSetup`, which should then be moved to a separate helper package.

File: feed_test.go (hunk 4 above)
Comment (reifcode) [reply feed_test.go]: as you can see, it has a different foreign id

File: feed_test.go (hunk 3 above)
Comment (reifcode) [reply feed_test.go]: I see your point. In this case it was just testing the coherence of the errors, but makes sense to just test the happy path.

File: feed_test.go (hunk 2 above)
Comment (reifcode) [reply feed_test.go]: actually, makes more sense to split the test in two and put both into flat feed tests and aggregated feeds tests ;)

File: feed_aggregated_test.go [hunk 5]
```diff
@@ -6,16 +6,18 @@ import (
...
 
 	getstream "github.com/GetStream/stream-go"
+	"github.com/stretchr/testify/assert"
 )
 
-func getAggregatedFeed(client *getstream.Client) *getstream.AggregatedFeed {
-	f, _ := client.AggregatedFeed("aggregated", RandString(8))
+func getAggregatedFeed(t *testing.T, client *getstream.Client) *getstream.AggregatedFeed {
+	f, err := client.AggregatedFeed("aggregated", RandString(8))
+	assert.Nil(t, err)
```
Comment (JelteF) [inline feed_aggregated_test.go]: You should use `require.NoError`, it prints the error better and it quits the test instantly. This should be done in other places as well.
